modules/connectors/dpd_group/karrio/providers/dpd_group/shipment/create.py

```diff
@@ -0,0 +1,165 @@
+"""Karrio DPD Group shipment API implementation."""
+
+import typing
+import karrio.lib as lib
+import karrio.core.units as units
+import karrio.core.models as models
+import karrio.providers.dpd_group.error as error
+import karrio.providers.dpd_group.utils as provider_utils
+import karrio.providers.dpd_group.units as provider_units
+import karrio.core.errors as errors
+import karrio.schemas.dpd_group.shipment_request as dpd_req
+import karrio.schemas.dpd_group.shipment_response as dpd_res
+
+
+def parse_shipment_response(
+    _response: lib.Deserializable[dict],
+    settings: provider_utils.Settings,
+) -> typing.Tuple[models.ShipmentDetails, typing.List[models.Message]]:
+    response = _response.deserialize()
+    messages = error.parse_error_response(response, settings)
+
+    # Check if we have valid shipment data
+    has_shipment = "shipmentId" in response if isinstance(response, dict) else False
+
+    shipment = _extract_details(response, settings) if has_shipment and not any(messages) else None
+
+    return shipment, messages
+
+
+def _extract_details(
+    data: dict,
+    settings: provider_utils.Settings,
+) -> models.ShipmentDetails:
+    """Extract shipment details from DPD META-API response."""
+    shipment = lib.to_object(dpd_res.ShipmentResponseType, data)
+
+    # Extract tracking number from parcelIds
+    tracking_number = shipment.parcelIds[0] if shipment.parcelIds else ""
+
+    # Extract label data
+    label_data = shipment.label.base64Data if shipment.label else None
+
+    # Build tracking URL
+    tracking_url = settings.tracking_url.format(tracking_number) if tracking_number else None
+
+    return models.ShipmentDetails(
+        carrier_id=settings.carrier_id,
+        carrier_name=settings.carrier_name,
+        tracking_number=tracking_number,
+        shipment_identifier=shipment.shipmentId,
+        label_type="PDF",  # DPD returns PDF by default
+        docs=models.Documents(label=label_data),
+        meta=dict(
+            network_shipment_id=shipment.networkShipmentId,
+            network_parcel_ids=shipment.networkParcelIds,
+            parcel_barcodes=lib.to_dict(shipment.parcelBarcodes) if shipment.parcelBarcodes else [],
+            tracking_url=tracking_url,
+        ),
+    )
+
+
+def shipment_request(
+    payload: models.ShipmentRequest,
+    settings: provider_utils.Settings,
+) -> lib.Serializable:
+    """Create a DPD META-API shipment request."""
+    # DPD META-API requires sender.customerInfos identifiers (Swagger: sender.customerInfos.customerID/customerAccountNumber).
+    # If we don't have these, DPD responds with errors like:
+    # - COMMON_7 generalShipmentData.sender.customerNumber
+    customer_id = (settings.account_number or "").strip()
+    customer_account = (settings.customer_account_number or "").strip() or customer_id
```

Comment:
Shouldn't be done here.

Assume clean! 


```suggestion
    customer_id = settings.account_number
    customer_account = settings.customer_account_number or customer_id
```